nautobot/utilities/testing/integration.py

```diff
@@ -0,0 +1,142 @@
+import os
+from unittest import skipIf
+
+from django.contrib.staticfiles.testing import StaticLiveServerTestCase
+from django.conf import settings
+from django.urls import reverse
+from django.utils.functional import classproperty
+from selenium import webdriver
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+# URL used to connect to the Selenium host
+SELENIUM_URL = os.getenv("NAUTOBOT_SELENIUM_URL", "http://localhost:4444/wd/hub")
+
+# Hostname used by Selenium client to talk to Nautobot
+SELENIUM_HOST = os.getenv("NAUTOBOT_SELENIUM_HOST", "localhost")
+
+# Default login URL
+LOGIN_URL = reverse(settings.LOGIN_URL)
+
+
+class NautobotRemote(webdriver.Remote):
+    """Custom WebDriver with helpers."""
+
+    def wait_for_html(self, tag_name, timeout=2):
+        """Wait for the page to render and make sure we find `tag_name`."""
+        WebDriverWait(self, timeout=timeout).until(lambda driver: driver.find_element_by_tag_name(tag_name))
+
+    def find_button(self, button_text):
+        """Return a `<button>` element with the given `button_text`."""
+        return self.find_element_by_xpath(f'//button[text()="{button_text}"]')
+
+
+@skipIf(
+    "NAUTOBOT_INTEGRATION_TEST" not in os.environ,
+    "NAUTOBOT_INTEGRATION_TEST environment variable not set",
+)
+class SeleniumTestCase(StaticLiveServerTestCase):
+    """
+    Base test case for Selenium integration testing with custom helper mthods.
+
+    This extends `django.contrib.staticfiles.testing.StaticLiveServerTestCase`
+    so there is no need to run `collectstatic` prior to running tests.
+    """
+
+    host = "0.0.0.0"  # Always listen publicly
+    selenium_host = SELENIUM_HOST  # Docker: `nautobot`; else `localhost`
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Selenium remote client
+        cls.selenium = NautobotRemote(
+            command_executor=SELENIUM_URL,
+            options=cls._create_firefox_options(),
+            browser_profile=cls._create_firefox_profile(),
+        )
+
+        # Wait for the DOM in case an element is not yet rendered.
+        cls.selenium.implicitly_wait(10)
+
+    @classproperty
+    def live_server_url(cls):
+        return "http://%s:%s" % (cls.selenium_host, cls.server_thread.port)
```

Comment:
```suggestion
        return f"http://{cls.selenium_host}:{cls.server_thread.port}"
```

Reply:
Oh what the heck, I had that in there and reverted it in my 753 commits to get Travis working. Thanks!